torch/_refs/__init__.py

```diff
@@ -436,7 +442,18 @@ def _ref(a: TensorLikeType) -> TensorLikeType:
             if extra_meta is not None:
                 extra_meta(a)
 
-            return prim(a)
+            out = prim(a)
+
+            device = None
+            from torch._subclasses.fake_tensor import FakeTensor
+
+            if isinstance(a, FakeTensor):
+                device = a.fake_device
+
+            if is_noncontiguous_supported(device) is False:
```

Comment:
`if not is_noncontiguous_supported(device)`

Reply:
done.